packages/backend-auth/src/factory.ts

```diff
@@ -130,6 +131,10 @@ class AmplifyAuthGenerator implements ConstructContainerEntryGenerator {
       ),
       outputStorageStrategy: this.getInstanceProps.outputStorageStrategy,
     };
+    if (authProps.loginWith.externalProviders) {
+      authProps.loginWith.externalProviders.domainPrefix =
+        stableBackendIdentifiers.getStableBackendHash();
```

Comment:
Is the consensus here that a hash of type, namespace, and name is sufficient to avoid collisions?
<img width="770" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/110861985/73b2b034-b1bf-4139-8291-0f0b30926021">

Reply:
yes, a combination of `type`, `namespace`, and `name` results in a unique stackname